src/content/docs/learn/window-menu.mdx

```diff
@@ -548,5 +569,372 @@ fn main() {
+submenu.setIcon('document');
+// Or set a native icon
+submenu.setNativeIcon('NSFolder');
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::{
+    image::Image,
+    menu::{Menu, MenuItem, SubmenuBuilder},
+};
+
+// Load icon from path
+let icon_image = Image::from_bytes(include_bytes!("../icons/icon.png")).unwrap();
+
+let submenu = SubmenuBuilder::new(app, "File")
+    .item(&MenuItem::with_id(app, "new", "New", true, None::<&str>)?)
+    .item(&MenuItem::with_id(app, "open", "Open", true, None::<&str>)?)
+    .icon(icon_image.clone()) // Add an icon to the submenu
+    .build()?;
+
+let menu = Menu::with_items(app, &[&submenu])?;
+app.set_menu(menu)?;
+
+// You can also update the submenu icon dynamically
+let icon_image_update = Image::from_bytes(include_bytes!("../icons/icon_update.png")).unwrap();
+submenu.set_icon(Some(icon_image_update))?;
+// Or set a native icon
+submenu.set_native_icon(Some(tauri::menu::NativeIcon::Folder))?;
+```
+
+</TabItem>
+</Tabs>
+
+### Menu Item States
+
+You can enable/disable menu items:
+
+<Tabs syncKey="menu-states">
+<TabItem label="JavaScript">
+
+```javascript
+import { Menu } from '@tauri-apps/api/menu';
+
+const menu = await Menu.new({
+  items: [
+    {
+      id: 'enabled',
+      text: 'Enabled Item',
+      enabled: true,
+    },
+    {
+      id: 'disabled',
+      text: 'Disabled Item',
+      enabled: false,
+    },
+  ],
+});
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::menu::{Menu, MenuItem};
+
+let enabled_item = MenuItem::with_id(app, "enabled", "Enabled Item", true, None::<&str>)?;
+let disabled_item = MenuItem::with_id(app, "disabled", "Disabled Item", false, None::<&str>)?;
+
+let menu = Menu::with_items(app, &[&enabled_item, &disabled_item])?;
+```
+
+</TabItem>
+</Tabs>
+
+## Advanced Menu Features
+
+### Dynamic Menu Updates
+
+You can update the menu dynamically:
+
+<Tabs syncKey="dynamic-menu">
+<TabItem label="JavaScript">
+
+```javascript
+import { Menu } from '@tauri-apps/api/menu';
+
+// Create initial menu
+const initialMenu = await Menu.new({
+  items: [
+    {
+      id: 'item1',
+      text: 'Initial Item',
+    },
+  ],
+});
+
+await initialMenu.setAsAppMenu();
+
+// Update menu later
+const updatedMenu = await Menu.new({
+  items: [
+    {
+      id: 'item2',
+      text: 'Updated Item',
+    },
+  ],
+});
+
+await updatedMenu.setAsAppMenu();
+```
+
+</TabItem>
+
+<TabItem label="Rust">
+
+```rust
+use tauri::menu::{Menu, MenuItem};
+
+// Create initial menu
+let initial_item = MenuItem::with_id(app, "item1", "Initial Item", true, None::<&str>)?;
+let initial_menu = Menu::with_items(app, &[&initial_item])?;
+
+app.set_menu(initial_menu)?;
+
+// Update menu later
+let updated_item = MenuItem::with_id(app, "item2", "Updated Item", true, None::<&str>)?;
+let updated_menu = Menu::with_items(app, &[&updated_item])?;
+
+app.set_menu(updated_menu)?;
+```
+
+</TabItem>
+</Tabs>
+
+### Updating Menu Item Text
```

Comment:
I think the examples inside this section are currently the same as the one above?